src/anyp/Uri.cc

```diff
@@ -571,10 +575,12 @@ AnyP::Uri::authority(bool requirePort) const
         authorityWithPort_.append(host());
         authorityHttp_ = authorityWithPort_;
 
-        // authorityForm_ only has :port if it is non-default
-        authorityWithPort_.appendf(":%u",port());
-        if (port() != getScheme().defaultPort())
-            authorityHttp_ = authorityWithPort_;
+        if (port().has_value()) {
+            authorityWithPort_.appendf(":%hu", port().value());
```

Comment:
Normally, we get here with a port number. If there are some special cases when we get here without the port number, the old code made `authorityWithPort_` look like `host:0`. This PR leaves authorityWithPort_ unchanged -- `host`. I am not 100% sure whether that is an improvement, but it may be -- we are no longer leaking a bogus port number for somebody to pick up. If you disagree, I will adjust this code to append zero port again.

A _proper_ fix would be to leave `authorityWithPort_` completely empty and throw if somebody actually attempts to _use_ it later: Callers should not ask for and should not be given "authority with port" when there is no port. However, that proper  fix is outside this PR scope and it requires serious auditing work (and possibly significant code adjustments).

Reply:
Done at 5d9fe29.

Reply:
We can simplify a little:
```suggestion
            authorityWithPort_.appendf(":%hu", *port());
```
since we already checked with `has_value()`.